plugins/pulp_rpm/plugins/importers/yum/config_validate.py

```diff
@@ -25,3 +46,69 @@ def validate(config):
             msg += failure_message + '\n'
         msg = msg.rstrip()  # remove the trailing \n
         return False, msg
+
+
+def _validate_allow_keys(repo, config, error_messages):
```

Comment:
I think this method should have an additional check. Effectively it should enforce the following pseudo code:

If the repo has > 0 units and this update is removing an existing allowed_keys then disallow the update.

In other words, once there is content in the repo additions to allow_keys is allowed, but removals of an existing key are not.